pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java

```diff
@@ -277,6 +277,12 @@ protected FileFormat getFileFormat(String fileName) {
     if (fileName.endsWith(".thrift")) {
       return FileFormat.THRIFT;
     }
+    if (fileName.endsWith(".parquet")) {
+      return FileFormat.PARQUET;
+    }
+    if (fileName.endsWith(".orc")) {
```

Comment:
Correct me if you are understanding differently / configuring it differently, but when I configure orc, I configure. Same with the comment I left yesterday in SegmentCreationJob.

record.reader.path: ${PATH_TO_ORCREADER} 
therefore, it skips the fileName.endsWith

    if (_recordReaderPath != null) {
      segmentGeneratorConfig.setRecordReaderPath(_recordReaderPath);
      segmentGeneratorConfig.setFormat(FileFormat.OTHER);
    } else {
      FileFormat fileFormat = getFileFormat(inputFileName);
      segmentGeneratorConfig.setFormat(fileFormat);
      segmentGeneratorConfig.setReaderConfig(getReaderConfig(fileFormat));
    }

Reply:
yes, so basically it requires extra configs in order to read from orc/parquet files

Reply:
We count ORC as FileFormat.OTHER and set the record reader path to the ORCRecordReader, so you don't need to include the orc-specific logic for segmentcreation to work